src/Pages/History/history.js

```diff
@@ -0,0 +1,88 @@
+import { Link, useNavigate } from "react-router-dom"
+import { useAuth } from "../../Contexts/auth-context"
+import axios from "axios"
+import {Header} from "../../Components/header"
+import { useEffect } from "react"
+import { useServices } from "../../Contexts/service-context"
+import { Sidebar } from "../../Components/sidebar"
+
+const HistoryPage = () => {
+    const { videoState , dispatchVideo} = useServices()
+    const { 
+        auth: { token  } ,
+      }  = useAuth()
+
+      const navigate = useNavigate()
+
+      useEffect(() => {
+        axios.get("/api/user/history", {headers : {authorization : token}}).then(res => {
+          dispatchVideo({type : "GET_HISTORY_VIDEOS" , payload : res.data.history})
+        })
+      },[])
+
+      const getSingleVideo = async  (video) => {
+        navigate(`/video/${video._id}`)
+    }
+    
+
+    const deleteFromHistoryService = async (video) => {
+        try {
+          await axios.delete(`/api/user/history/${video._id}` , {headers :{authorization : token} , });
+        } catch(e) {
+          console.log(e)
+        }
+      }
+      
+      const deleteAllFromHistoryService = async () => {
+        try {
+          await axios.delete("/api/user/history/all", {headers : {authorization : token}})
+        } catch(e){
+          console.log(e)
+        }
+      }
+
+    const deleteAllHistoryVideos = async (token) => {
+        await deleteAllFromHistoryService(token);
+        dispatchVideo({type : "DELETE_ALL_HISTORY_VIDEOS"})
+      }
+      
+      const deleteFromHistory = async (video , token) => {
+        await deleteFromHistoryService(video , token);
+        dispatchVideo({type : "DELETE_HISTORY_VIDEO" , payload : video})
+      }
+      
```

Comment:
you can move this code to utils or services

Reply:
Will extract all the utils at the end.